lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -239,7 +239,11 @@ def filter_query(filters: list[str], query: Query) -> Query:
                 else:
                     try:
                         c = getattr(Dataset, second)
-                        if c.type.python_type is str:
+                        try:
+                            is_str = c.type.python_type is str
+                        except NotImplementedError:
+                            is_str = False
```

Comment:
Is trying to reference `c.type.python_type` better than using the [`type()` builtin](https://docs.python.org/3/library/functions.html#type), e.g.?
```
                        is_str = type(c) is str
```
Or, perhaps the recommended approach of
```
                        is_str = isinstance(c, str)
```
I expect that either of these would deliver you from having to use a `try` block....